src/hb-common.cc

```diff
@@ -35,10 +35,6 @@
 #include <xlocale.h>
 #endif
 
-#ifdef HB_NO_SETLOCALE
-#define setlocale(Category, Locale) "C"
-#endif
```

Comment:
Apparently redefining this way is not enough, that motivated to look for another solution and thought lets turn it into a check anyway.